pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/BaseSingleTreeBuilder.java

```diff
@@ -520,11 +520,22 @@ private void createForwardIndexes()
         }
       }
     } finally {
-      for (SingleValueUnsortedForwardIndexCreator dimensionIndexCreator : dimensionIndexCreators) {
-        dimensionIndexCreator.close();
+      for (int i = 0; i < dimensionIndexCreators.length; i++) {
+        try {
+          dimensionIndexCreators[i].close();
+        } catch (Throwable e) {
+          LOGGER.warn("Caught throwable while closing dimension index creator for dimension: {}",
+              _dimensionsSplitOrder[i], e);
+          throw e;
```

Comment:
throwing inside a finally section is an anti-pattern, but in fact we were already doing that before this change.

Reply:
Does `close()` throw exception when the buffer size is over the integer bound? Should we cache the first exception to ensure all creators are closed? Currently if it throws exception, we will leave creators not closed

Reply:
I guess that would be the best solution, yes. I'm going to change this code